clientUtils/patchHelper.ts

```diff
@@ -1,7 +1,25 @@
 import jsonpointer from "json8-pointer"
 import { isNil } from "lodash"
 import { GrapherConfigPatch } from "./AdminSessionTypes.js"
-import { isArray, isPlainObjectWithGuard } from "./Util.js"
+import { isArray, isEqual, isPlainObjectWithGuard } from "./Util.js"
+
+export function setValueRecursiveInplace(
+    json: any,
+    pointer: string[],
+    newValue: any
+): any {
+    if (pointer.length === 0) throw new Error("Pointer must not be empty")
+    const key: string = pointer.shift()!
+    if (pointer.length === 0) {
+        json[key] = newValue
+        return json
+    }
+    if (!json[key]) {
+        json[key] = {}
+    }
+    return setValueRecursiveInplace(json[key], pointer, newValue)
```

Comment:
A bit worried about these edge cases: 
- if it's non-empty and a falsy value (`""`, `0`, `false`) it gets overwritten with `{}`
- if it's non-empty and a truthy value (`"abc"`, `1`), then we attempt to inject a property (even when it's a primitive, which becomes a noop)

It seems very unlikely they show up, but if they do, maybe better to raise an error? I think the only case we should set `{}` is when `json[key] === undefined`?

Reply:
Good catch - this code was suggested by github copilot and I was so blown away by it that I accepted it after only checking some common edge cases. I now made a couple of things more explicit (esp array handling) and added tests